### DISCUSSION TITLE
Room cleaning as separate accessories

### DISCUSSION THREAD

**Discussion by Seppo47** [homebridge-xiaomi-roborock-vacuum/homebridge-xiaomi-roborock-vacuum#467] [Ideas]
First of all thanks for the great plugin, works great with my Roborock S7 + Homebridge + Homekit.

I've setup room specific cleaning as separate switches and those work great in Homekit. However, because the switches are all part of the same vacuum accessory, even if I can split them as separate tiles in Home app, I'm not able to assign them to different rooms. My ideal setup would be that e.g. "clean livingroom" switch would be in "livingroom" in my home app, "clean bedroom" in "bedroom" etc. Now they all need to be in the same room.

So I was wondering if you could add a config parameter to make the room cleaning switches to be published as separate accessories. This would enable me to assign the switches to their respective rooms. I'd imagine that other Roborock+Homekit users would also appreciate this feature. (or of course if this is already possible with some trick, please let me know)

Big thanks again for your great work.

**Comment by afharo** [comment]
Thank you for opening this discuss item and for your suggestion. I completely agree! It's ugly and it doesn't make sense to have the room switches in a different room.

My idea is to convert this plugin into a Platform, so it can create accessories at will. 
This will also solve some issues like only registering the services when the vacuum is connected and we find that they are supported by the specific model.

I can only work slowly towards this goal. But that's the future plan.
